src/clp_ffi_js/ir/StreamReaderContext.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef CLP_FFI_JS_IR_STREAM_READER_CONTEXT_HPP
+#define CLP_FFI_JS_IR_STREAM_READER_CONTEXT_HPP
+
+#include <memory>
+#include <utility>
+
+#include <clp/Array.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/streaming_compression/zstd/Decompressor.hpp>
+
+namespace clp_ffi_js::ir {
+/**
+ * The data context for a `StreamReader`. It encapsulates a chain of the following resources:
+ * A `clp::ir::LogEventDeserializer` that reads from a
+ * `clp::streaming_compression::zstd::Decompressor`, which in turn reads from a `clp::Array`.
+ * @tparam encoded_variable_t Type of encoded variables encoded in the stream.
+ */
+template <typename encoded_variable_t>
+class StreamReaderContext {
+public:
+    // Constructors
+    explicit StreamReaderContext(
+            clp::Array<char>&& data_buffer,
+            std::unique_ptr<clp::streaming_compression::zstd::Decompressor>&& zstd_decompressor,
+            clp::ir::LogEventDeserializer<clp::ir::four_byte_encoded_variable_t> deserializer
+    )
+            : m_data_buffer{std::make_unique<clp::Array<char>>(std::move(data_buffer))},
```

Comment:
Does `m_data_buffer` need to be a `unique_ptr`?

Reply:
No need for now. sorry for missing this when I ported the code over